metabase/metabase
# Refactor CheckBox and StackedCheckBox components

This PR refactors `CheckBox` and `StackedCheckBox` components. The main motivation is to fix a11y issues, but it also aligns components with our current code style.

* refactors `CheckBox` to use `<input type="checkbox" />` under the hood. Currently it's just a div with `role="checkbox"` and click handlers
* refactors `StackedCheckBox` to use only one `CheckBox` input, and to render a stacked background as a different component. Currently, it just renders two checkboxes, so if you tab through the page, you'd focus on both of them which is pretty unexpected
* extends both components with a `label` prop. Now you can pass a string or a custom React component to display a label on the right side (clicking a label would lead to a `checked` change as if you click the checkbox itself)
* refactors `CheckBox` not to use deprecated color names
* code style: migrated to styled-components, prop-types position, .etc

### To Verify

1. Open `/_internal/components/checkbox` and `/_internal/components/stackedcheckbox`
2. Ensure every demo makes sense

Note: `StackedCheckBox` with a label renders weirdly in the StyleGuide, but looks as expected in the app itself. You can check a `FieldsPicker` example

1. Ask a question > Custom Question > Sample Dataset > Any Table
2. Click a "Columns" button on the right side of the "Data" panel (the one with a data source)
3. A popup with a list of table columns opens, look on the "Select All" checkbox in the top
